src/main/java/kr/codesqaud/cafe/controller/ArticleController.java

```diff
@@ -37,7 +45,35 @@ public String write(@ModelAttribute ArticleSaveRequest articleSaveRequest) {
 
     @GetMapping("/articles/{id}")
     public String getArticle(@PathVariable Long id, Model model) {
-        model.addAttribute("article", articleService.findById(id));
+        model.addAttribute("article", articleService.getArticleWithSurrounding(id));
         return "article/detail";
     }
+
+    @GetMapping("/articles/{id}/edit")
+    public String editArticle(@PathVariable Long id, Model model, HttpSession httpSession) {
+        final ArticleResponse article = articleService.findById(id);
+        if (!httpSession.getAttribute(SessionAttributeNames.LOGIN_USER_NAME).equals(article.getWriter())) {
```

Comment:
코드가 반복되고 있네요. 다음과 같이 공통 메소드로 합치는 것을 고려해보세요.
```suggestion
    checkUserPermission(httpSession, articleUpdateRequest.getWriter());
```
```java
private void checkUserPermission(HttpSession httpSession, String writer) {
     if (!httpSession.getAttribute(SessionAttributeNames.LOGIN_USER_NAME).equals(writer)) {
         throw new AccessDeniedException();
     }
}
```

Reply:
UserController에서 반복되는 코드도 함께 분리했습니다